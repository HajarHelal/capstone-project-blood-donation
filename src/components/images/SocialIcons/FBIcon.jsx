import React from 'react';

function FBIcon() {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="35.68"
      height="32.74"
      fill="none"
      viewBox="0 0 18 22"
      className="mr-4 h-6 w-6 transform hover:scale-125 transition duration-500 ease-out"
    >
      <path
        fill="#fff"
        fillRule="evenodd"
        d="M9.72 20.242v-9.527h2.025l.269-3.283H9.719l.004-1.643c0-.856.062-1.315 1.01-1.315h1.266V1.19H9.973c-2.434 0-3.29 1.593-3.29 4.271v1.971H5.166v3.283h1.517v9.527h3.036z"
        clipRule="evenodd"
      />
      <mask width="8" height="20" x="5" y="1" maskUnits="userSpaceOnUse">
        <path
          fill="#fff"
          fillRule="evenodd"
          d="M9.72 20.242v-9.527h2.025l.269-3.283H9.719l.004-1.643c0-.856.062-1.315 1.01-1.315h1.266V1.19H9.973c-2.434 0-3.29 1.593-3.29 4.271v1.971H5.166v3.283h1.517v9.527h3.036z"
          clipRule="evenodd"
        />
      </mask>
    </svg>
  );
}

export default FBIcon;
