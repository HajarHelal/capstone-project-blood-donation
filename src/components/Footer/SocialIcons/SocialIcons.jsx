import React from 'react';
import FBIcon from '../../images/SocialIcons/FBIcon';
import InstaIcon from '../../images/SocialIcons/InstaIcon';
import LinkedInIcon from '../../images/SocialIcons/LinkedInIcon';
import TwitterIcon from '../../images/SocialIcons/TwitterIcon';
import UtupeIcon from '../../images/SocialIcons/UtupeIcon';

function SocialIcons() {
  return (
    <div className="flex my-8">
      <a target="_blank" rel="noreferrer" href="http://www.facebook.com">
        <FBIcon />
      </a>
      <a target="_blank" rel="noreferrer" href="https://www.instagram.com/">
        <InstaIcon />
      </a>
      <a target="_blank" rel="noreferrer" href="https://www.linkedin.com/">
        <LinkedInIcon />
      </a>
      <a target="_blank" rel="noreferrer" href="https://twitter.com/">
        <TwitterIcon />
      </a>
      <a target="_blank" rel="noreferrer" href="https://www.youtube.com/">
        <UtupeIcon className="mr-4" />
      </a>
    </div>
  );
}
export default SocialIcons;
