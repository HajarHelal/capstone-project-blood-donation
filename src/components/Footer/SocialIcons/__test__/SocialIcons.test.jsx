import React from 'react';
import renderer from 'react-test-renderer';
import SocialIcons from '../SocialIcons';

// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer.create(<SocialIcons />).toJSON();
  expect(tree).toMatchSnapshot();
});