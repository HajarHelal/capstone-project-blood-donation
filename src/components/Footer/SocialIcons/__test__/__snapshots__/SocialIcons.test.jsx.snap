// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex my-8"
>
  <a
    href="http://www.facebook.com"
    rel="noreferrer"
    target="_blank"
  >
    <svg
      className="mr-4 "
      fill="none"
      height="32.74"
      viewBox="0 0 18 22"
      width="35.68"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M9.72 20.242v-9.527h2.025l.269-3.283H9.719l.004-1.643c0-.856.062-1.315 1.01-1.315h1.266V1.19H9.973c-2.434 0-3.29 1.593-3.29 4.271v1.971H5.166v3.283h1.517v9.527h3.036z"
        fill="#fff"
        fillRule="evenodd"
      />
      <mask
        height="20"
        maskUnits="userSpaceOnUse"
        width="8"
        x="5"
        y="1"
      >
        <path
          clipRule="evenodd"
          d="M9.72 20.242v-9.527h2.025l.269-3.283H9.719l.004-1.643c0-.856.062-1.315 1.01-1.315h1.266V1.19H9.973c-2.434 0-3.29 1.593-3.29 4.271v1.971H5.166v3.283h1.517v9.527h3.036z"
          fill="#fff"
          fillRule="evenodd"
        />
      </mask>
    </svg>
  </a>
  <a
    href="https://www.instagram.com/"
    rel="noreferrer"
    target="_blank"
  >
    <svg
      className="mr-4"
      fill="none"
      height="32.74"
      viewBox="0 0 17 22"
      width="35.68"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M8.01 1.193h-.505V1.19c-1.668.003-1.988.017-2.814.066-.873.052-1.347.24-1.662.4-.418.212-.716.463-1.03.87a3.838 3.838 0 00-.67 1.336c-.122.41-.268 1.025-.308 2.158C.978 7.245.97 7.61.97 10.714c0 3.102.008 3.47.051 4.695.04 1.133.186 1.748.308 2.157.163.543.357.93.67 1.336.314.406.612.658 1.03.869.316.159.79.348 1.662.4.944.056 1.227.068 3.617.068 2.39 0 2.674-.012 3.617-.068.873-.051 1.347-.24 1.662-.4.418-.21.716-.462 1.03-.869a3.83 3.83 0 00.669-1.335c.123-.409.268-1.024.308-2.157.043-1.225.053-1.593.053-4.694 0-3.1-.01-3.469-.053-4.693-.04-1.133-.185-1.748-.308-2.157a3.844 3.844 0 00-.67-1.337c-.313-.406-.61-.658-1.029-.869-.315-.159-.789-.348-1.662-.4-.943-.056-1.227-.067-3.617-.067H8.01zm4.002 3.324c0-.77.481-1.394 1.074-1.394.593 0 1.075.624 1.075 1.394 0 .77-.482 1.394-1.075 1.394s-1.074-.625-1.074-1.394zm-3.704.235c-2.539 0-4.597 2.672-4.597 5.967 0 3.295 2.058 5.966 4.597 5.966 2.539 0 4.596-2.67 4.596-5.966 0-3.295-2.057-5.967-4.596-5.967zm2.984 5.967c0-2.14-1.336-3.873-2.984-3.873S5.324 8.58 5.324 10.719c0 2.14 1.336 3.873 2.984 3.873s2.984-1.734 2.984-3.873z"
        fill="#fff"
        fillRule="evenodd"
      />
      <mask
        height="20"
        maskUnits="userSpaceOnUse"
        width="16"
        x="0"
        y="1"
      >
        <path
          clipRule="evenodd"
          d="M8.01 1.193h-.505V1.19c-1.668.003-1.988.017-2.814.066-.873.052-1.347.24-1.662.4-.418.212-.716.463-1.03.87a3.838 3.838 0 00-.67 1.336c-.122.41-.268 1.025-.308 2.158C.978 7.245.97 7.61.97 10.714c0 3.102.008 3.47.051 4.695.04 1.133.186 1.748.308 2.157.163.543.357.93.67 1.336.314.406.612.658 1.03.869.316.159.79.348 1.662.4.944.056 1.227.068 3.617.068 2.39 0 2.674-.012 3.617-.068.873-.051 1.347-.24 1.662-.4.418-.21.716-.462 1.03-.869a3.83 3.83 0 00.669-1.335c.123-.409.268-1.024.308-2.157.043-1.225.053-1.593.053-4.694 0-3.1-.01-3.469-.053-4.693-.04-1.133-.185-1.748-.308-2.157a3.844 3.844 0 00-.67-1.337c-.313-.406-.61-.658-1.029-.869-.315-.159-.789-.348-1.662-.4-.943-.056-1.227-.067-3.617-.067H8.01zm4.002 3.324c0-.77.481-1.394 1.074-1.394.593 0 1.075.624 1.075 1.394 0 .77-.482 1.394-1.075 1.394s-1.074-.625-1.074-1.394zm-3.704.235c-2.539 0-4.597 2.672-4.597 5.967 0 3.295 2.058 5.966 4.597 5.966 2.539 0 4.596-2.67 4.596-5.966 0-3.295-2.057-5.967-4.596-5.967zm2.984 5.967c0-2.14-1.336-3.873-2.984-3.873S5.324 8.58 5.324 10.719c0 2.14 1.336 3.873 2.984 3.873s2.984-1.734 2.984-3.873z"
          fill="#fff"
          fillRule="evenodd"
        />
      </mask>
    </svg>
  </a>
  <a
    href="https://www.linkedin.com/"
    rel="noreferrer"
    target="_blank"
  >
    <svg
      className="mr-4"
      fill="none"
      height="32.74"
      viewBox="0 0 18 22"
      width="35.68"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M4.982 3.31c-.02-1.203-.683-2.12-1.759-2.12-1.076 0-1.78.917-1.78 2.12 0 1.178.683 2.12 1.74 2.12h.02c1.096 0 1.78-.942 1.78-2.12zm-.207 3.795H1.63v12.269h3.146V7.104zm7.723-.288c2.07 0 3.622 1.754 3.622 5.522v7.035h-3.146V12.81c0-1.649-.455-2.774-1.594-2.774-.869 0-1.386.759-1.614 1.491-.083.263-.103.629-.103.995v6.852H6.516s.042-11.117 0-12.269h3.147v1.738c.417-.835 1.165-2.026 2.835-2.026z"
        fill="#fff"
        fillRule="evenodd"
      />
      <mask
        height="19"
        maskUnits="userSpaceOnUse"
        width="16"
        x="1"
        y="1"
      >
        <path
          clipRule="evenodd"
          d="M4.982 3.31c-.02-1.203-.683-2.12-1.759-2.12-1.076 0-1.78.917-1.78 2.12 0 1.178.683 2.12 1.74 2.12h.02c1.096 0 1.78-.942 1.78-2.12zm-.207 3.795H1.63v12.269h3.146V7.104zm7.723-.288c2.07 0 3.622 1.754 3.622 5.522v7.035h-3.146V12.81c0-1.649-.455-2.774-1.594-2.774-.869 0-1.386.759-1.614 1.491-.083.263-.103.629-.103.995v6.852H6.516s.042-11.117 0-12.269h3.147v1.738c.417-.835 1.165-2.026 2.835-2.026z"
          fill="#fff"
          fillRule="evenodd"
        />
      </mask>
    </svg>
  </a>
  <a
    href="https://twitter.com/"
    rel="noreferrer"
    target="_blank"
  >
    <svg
      className="mr-4"
      fill="none"
      height="32.74"
      viewBox="0 0 17 22"
      width="35.68"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M8.043 6.57l.032.68-.53-.083c-1.93-.32-3.616-1.404-5.048-3.224l-.7-.903-.18.667c-.382 1.487-.138 3.057.657 4.113.425.584.33.667-.403.32-.254-.111-.477-.195-.498-.153-.074.097.18 1.362.382 1.862a4.014 4.014 0 001.453 1.779l.52.32-.616.014c-.594 0-.615.013-.551.305.212.904 1.05 1.863 1.983 2.28l.657.291-.572.445c-.849.64-1.846 1-2.842 1.028-.478.014-.87.07-.87.112 0 .139 1.294.917 2.047 1.223 2.259.903 4.942.514 6.956-1.029 1.432-1.098 2.864-3.28 3.532-5.392.36-1.126.72-3.182.72-4.17 0-.638.033-.722.627-1.486.35-.445.678-.931.742-1.07.106-.264.095-.264-.445-.028-.902.417-1.03.361-.584-.264.329-.445.721-1.25.721-1.487 0-.042-.159.028-.339.153-.19.139-.615.347-.933.472l-.573.237-.52-.46c-.286-.25-.689-.527-.9-.61-.542-.195-1.369-.167-1.857.055-1.325.625-2.163 2.237-2.068 4.002z"
        fill="#fff"
        fillRule="evenodd"
      />
      <mask
        height="17"
        maskUnits="userSpaceOnUse"
        width="16"
        x="0"
        y="2"
      >
        <path
          clipRule="evenodd"
          d="M8.043 6.57l.032.68-.53-.083c-1.93-.32-3.616-1.404-5.048-3.224l-.7-.903-.18.667c-.382 1.487-.138 3.057.657 4.113.425.584.33.667-.403.32-.254-.111-.477-.195-.498-.153-.074.097.18 1.362.382 1.862a4.014 4.014 0 001.453 1.779l.52.32-.616.014c-.594 0-.615.013-.551.305.212.904 1.05 1.863 1.983 2.28l.657.291-.572.445c-.849.64-1.846 1-2.842 1.028-.478.014-.87.07-.87.112 0 .139 1.294.917 2.047 1.223 2.259.903 4.942.514 6.956-1.029 1.432-1.098 2.864-3.28 3.532-5.392.36-1.126.72-3.182.72-4.17 0-.638.033-.722.627-1.486.35-.445.678-.931.742-1.07.106-.264.095-.264-.445-.028-.902.417-1.03.361-.584-.264.329-.445.721-1.25.721-1.487 0-.042-.159.028-.339.153-.19.139-.615.347-.933.472l-.573.237-.52-.46c-.286-.25-.689-.527-.9-.61-.542-.195-1.369-.167-1.857.055-1.325.625-2.163 2.237-2.068 4.002z"
          fill="#fff"
          fillRule="evenodd"
        />
      </mask>
    </svg>
  </a>
  <a
    href="https://www.youtube.com/"
    rel="noreferrer"
    target="_blank"
  >
    <svg
      className="mr-4"
      fill="none"
      height="32.74"
      viewBox="0 0 17 22"
      width="35.68"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M5.689 1.34h1.054L5.95 4.434a14.1 14.1 0 01-.145.516 28.6 28.6 0 00-.151.525 4.5 4.5 0 00-.109.516v3.3h-1.04v-3.15c-.02-.15-.159-.682-.419-1.594L3.261 1.34h1.055l.665 3.17h.072l.636-3.17zm3.683 3.45V7.83c0 .525-.12.925-.36 1.2-.242.275-.57.412-.983.412-.395 0-.715-.146-.961-.44-.246-.294-.369-.691-.369-1.191V4.772c0-.488.133-.86.398-1.116.265-.256.59-.384.975-.384.366 0 .674.137.925.412.25.275.375.644.375 1.107zm3.425-1.368v5.87h-.925v-.657c-.356.488-.713.731-1.07.731-.432 0-.65-.35-.65-1.05V3.422h.925v4.5c0 .263.082.394.246.394.164 0 .347-.15.55-.45V3.422h.924zM8.433 4.715v3.207c0 .325-.134.488-.404.488-.25 0-.376-.163-.376-.488V4.715c0-.125.039-.228.116-.31a.351.351 0 01.26-.121c.096 0 .188.04.274.122.087.081.13.184.13.31zm5.996 13.203c.058-1.2.087-2.107.087-2.72 0-.612-.03-1.518-.087-2.719-.039-.625-.224-1.122-.556-1.49a1.707 1.707 0 00-1.221-.591c-.944-.063-2.374-.094-4.29-.094-1.918 0-3.348.031-4.292.094a1.707 1.707 0 00-1.22.59c-.333.37-.518.866-.557 1.491a61.55 61.55 0 00-.086 2.72c0 .612.028 1.519.086 2.719.039.625.224 1.122.556 1.49.333.37.74.567 1.221.592.944.062 2.374.093 4.291.093 1.907 0 3.337-.03 4.291-.093a1.707 1.707 0 001.22-.591c.333-.369.518-.866.557-1.491zm-.94-1.013v-.356h-.765c0 .025.002.106.007.244.005.137.005.234 0 .29a1.75 1.75 0 01-.036.216c-.02.088-.053.15-.101.188a.322.322 0 01-.202.056.258.258 0 01-.224-.113.438.438 0 01-.087-.225 5.342 5.342 0 01-.007-.318v-.9h1.416v-1.2c0-.438-.092-.782-.275-1.032-.183-.25-.443-.375-.78-.375-.327 0-.6.128-.816.384-.217.256-.325.597-.325 1.022v2.12c0 .45.094.809.281 1.078.188.268.456.403.802.403.742 0 1.113-.494 1.113-1.482zm-2.758-2.269v2.626c0 .7-.227 1.05-.68 1.05-.26 0-.495-.144-.707-.431v.375h-.751v-6.433h.75v2.063c.203-.288.425-.431.665-.431.482 0 .723.393.723 1.181zm-3.612 3.62h.737V13.51h-.737v3.6c-.164.238-.313.357-.448.357-.125 0-.188-.106-.188-.319v-3.638h-.751v3.994c0 .55.173.826.52.826.289 0 .578-.2.867-.6v.524zM5.746 12.76h-.881v5.495h-.838V12.76H3.16v-.938h2.586v.938zm6.978 1.969v.525h-.65v-.525c0-.338.11-.507.332-.507.212 0 .318.17.318.507zm-2.76 2.438v-2.494c0-.313-.1-.47-.303-.47-.115 0-.221.063-.317.188v2.963c.106.138.226.207.36.207.174 0 .26-.132.26-.394z"
        fill="#fff"
        fillRule="evenodd"
      />
    </svg>
  </a>
</div>
`;
